Assets/_Challenges/Scripts/AI Behaviours/Debug/LogBehaviour.cs

```diff
@@ -31,21 +31,23 @@ public override void Tick(Chalkboard chalkboard)
             for (int i = 0; i < matches.Count; i ++)
             {
                 string token = matches[i].Groups[0].Value;
+                int index = matches[i].Groups[0].Index;
                 string variableName = matches[i].Groups[1].Value;
 
                 UnityEngine.Object unityValue = chalkboard.GetUnity<UnityEngine.Object>(variableName.GetHashCode());
                 if (unityValue != null)
                 {
-                    expandedMessage = expandedMessage.Replace(token, unityValue.ToString());
+                    expandedMessage = expandedMessage.Remove(index, token.Length).Insert(index, unityValue.ToString());
```

Comment:
Ah, not quite what I meant. I was thinking more along the lines of:
StringBuilder newString;
newString.append(0, clippedContent.index);
newString.append(variable)
...
Remove/Insert here may not be any more efficient than replace. Depending on how c#'s string internals work it may be doing a bunch of frees/allocs with Replace and This method. Given this is debug only you can leave it as is if you want.

Reply:
I did some investigation and this seemed to be the recommended method. However, that was generic c# guidance rather than Unity development - i.e. less sensitive to garbage collection.

I think we'll leave as is. Like you say it's just debug code and should never be in a final release. I strongly suspect this code will be revisited anyway as I'm thinking that allowing variable operations here would be very helpful. I find debugging behaviour trees really hard and being able to output logs with operations in them would be excellent (breakpoint and trace is problematic with Node Canvas and likely other BT implementations.)